File: 2015/1/1-2.py
with open('input.txt') as f:
  line = f.readline().strip('\n')
  pos = 0
  idx = 1
  for c in line:
    if c == '(':
      pos += 1
    if c == ')':
      pos -= 1
    if pos < 0:
      print ('Entered Basement at position %d' % (idx, ))
      break
    idx += 1

File: 2015/1/1-2.go
package main

import (
    "fmt"
    "os"
)

func main() {
    instructions := getFileLines("input.txt")[0]
    floor := 0

    for i, c := range instructions {
        if c == '(' {
            floor += 1
        }
        if c == ')' {
            floor -= 1
        }
        if floor == -1 {
            fmt.Printf("Entered basement at position %v\n", i + 1)
            os.Exit(0)
        }
    }
}
